app/views/cloud_volume/_common_new_edit.html.haml

```diff
@@ -35,26 +86,38 @@
   .col-md-8
     %input.form-control{:type          => "text",
                         :name          => "size",
-                        'ng-model'     => "cloudVolumeModel.size",
+                        'ng-model'     => "vm.cloudVolumeModel.size",
                         'ng-maxlength' => 10,
-                        'ng-change'    => "sizeChanged(cloudVolumeModel.size)",
-                        'ng-disabled'  => "formId != 'new' && (cloudVolumeModel.emstype != 'ManageIQ::Providers::Amazon::StorageManager::Ebs' || cloudVolumeModel.aws_volume_type == 'standard')",
+                        'ng-change'    => "sizeChanged(vm.cloudVolumeModel.size)",
+                        'ng-disabled'  => "!canModifyVolumeSize()",
                         :required      => "",
                         :checkchange   => true}
     %span.help-block{"ng-show" => "angularForm.size.$error.required"}
       = _("Required")
 
 .form-group{"ng-class" => "{'has-error': angularForm.aws_iops.$invalid}",
-            "ng-if"    => "cloudVolumeModel.emstype == 'ManageIQ::Providers::Amazon::StorageManager::Ebs'"}
+            "ng-if"    => "vm.cloudVolumeModel.emstype == 'ManageIQ::Providers::Amazon::StorageManager::Ebs'"}
   %label.col-md-2.control-label
     = _('IOPS')
   .col-md-8
     %input.form-control{:type          => "text",
                         :name          => "aws_iops",
-                        'ng-model'     => "cloudVolumeModel.aws_iops",
+                        'ng-model'     => "vm.cloudVolumeModel.aws_iops",
                         'ng-maxlength' => 50,
-                        'ng-disabled'  => "cloudVolumeModel.aws_volume_type != 'io1'",
-                        "ng-required"  => "cloudVolumeModel.aws_volume_type == 'io1'",
+                        'ng-disabled'  => "vm.cloudVolumeModel.aws_volume_type != 'io1'",
+                        "ng-required"  => "vm.cloudVolumeModel.aws_volume_type == 'io1'",
                         :checkchange   => true}
-    %span.help-block{"ng-show" => "cloudVolumeModel.aws_volume_type == 'io1' && angularForm.aws_iops.$error.required"}
+    %span.help-block{"ng-show" => "vm.cloudVolumeModel.aws_volume_type == 'io1' && angularForm.aws_iops.$error.required"}
       = _("Required")
+
+.form-group{"ng-if" => "vm.cloudVolumeModel.emstype == 'ManageIQ::Providers::Amazon::StorageManager::Ebs'"}
+  %label.col-md-2.control-label
+    = _('Encryption')
+  .col-md-8
+    %input.form-control{"bs-switch"   => "",
+                        :data         => {:on_text => _('Yes'), :off_text => _('No'), :size => 'mini'},
+                        :type         => "checkbox",
+                        :name         => "encryption",
+                        'ng-model'    => "vm.cloudVolumeModel.aws_encryption",
+                        'ng-disabled' => "!vm.newRecord",
```

Comment:
True, this was something I forgot to ask you, what is the correct way to disable the switch. Thanks for finding this out: https://github.com/ManageIQ/manageiq-ui-classic/pull/950/commits/1089882e4723bc63ccd5d2518777a6b7495d22d9